### PR TITLE
Added optional eventName parameter

### PR DESCRIPTION
**Pull request by erickjtorres**
Added an optional eventName parameter in the Lumberdashclient so the developer can specify the name of the event if they would so choose to. This is specifically useful for firebase since you can log multiple events and as a developer, I would like to have the ability to name all my events differently.

Feedback is appreciated

### CODE REVIEW

**Review by noahmateen — APPROVED** [review]

**Review by tchabot22 — APPROVED** [review]

File: lib/src/client/lumberdash_client.dart [hunk 1]
```diff
@@ -4,15 +4,15 @@ abstract class LumberdashClient {
   /// `message` is the most important piece of the log, and `extras`
   /// are attached information that can be added to expand the meaning
   /// and context of your log.
```
Comment (felangel) [inline lib/src/client/lumberdash_client.dart:7]: can we update the documentation above to explain the role of `eventName`?

**Review by felangel — CHANGES_REQUESTED** [review]
I think we should update the documentation

**Comment by XinfengMa** [comment]
I don't think eventName should be optional though because for firebase analytics logEvent name is a required: https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.html#logEvent(java.lang.String,%20android.os.Bundle)
also I think parameter message is an optional, it could be part of extras with key "message"

**Review by noahmateen — CHANGES_REQUESTED** [review]
Based on Xinfengs feedback, lets make that change

**Comment by jorgecoca** [comment]
@XinfengMa @noahmateen Since lumberdash is an abstraction over other libraries, we cannot let implementation details of other frameworks, such as firebase, dictate the API of the base client.
Maybe the naming could be improved, but adding the event name as not optional just because we need it for firebase is not the right answer, since other clients, such as sentry, or the simple client don't need it. That's why I think an optional parameter is a better option.

**Comment by XinfengMa** [comment]
@jorgecoca Just like you mentioned, lumberdash is abstract layer, seems like abstract a required parameter to an optional one is wrong approach.  I totally understand if we abstract an optional parameter to a required one, not other way around though.  Anyway I don't think this is the place to make the change.  Look at the code of FirebaseLumberdash, loggerName is used to log eventName there: https://pub.dev/documentation/firebase_lumberdash/latest/firebase_lumberdash/FirebaseLumberdash/logMessage.html

**Comment by erickjtorres** [comment]
> @jorgecoca Just like you mentioned, lumberdash is abstract layer, seems like abstract a required parameter to an optional one is wrong approach. I totally understand if we abstract an optional parameter to a required one, not other way around though. Anyway I don't think this is the place to make the change. Look at the code of FirebaseLumberdash, loggerName is used to log eventName there: https://pub.dev/documentation/firebase_lumberdash/latest/firebase_lumberdash/FirebaseLumberdash/logMessage.html

@XinfengMa I think the problem is that it's not always required and so we want to keep it optional. It also prevents us from breaking the current API for anyone that is currently using this. In the case of FirebaseLumberdash I'm creating a follow up PR to change the loggerName to being the value it defaults to if eventName is not provided.

**Comment by XinfengMa** [comment]
@erickjtorres just felt the abstraction is more tuned to Sentry rather Firebase.  Don't think it is a good abstraction.  logEvent is very common among most of logging frameworks, maybe create a new method?  Anyway if everybody else is OK with the current approach, I won't bee the holdout.

**Comment by erickjtorres** [comment]
> @erickjtorres just felt the abstraction is more tuned to Sentry rather Firebase. Don't think it is a good abstraction. logEvent is very common among most of logging frameworks, maybe create a new method? Anyway if everybody else is OK with the current approach, I won't bee the holdout.

I'm okay changing the approach to a proposed one or leaving it as is and coming up with a better approach later. Whatever the team feels comfortable with

**Comment by jorgecoca** [comment]
Ok, team, here's the approach we are going to take: I am going to close this PR, and let's plan for a V2. I am going to open an issue on Github where we can discuss what the API should be knowing what we know already about all the clients.

On Monday, after the design phase, we can work on making the changes.

Thank you!

**Comment by jorgecoca** [comment]
Let's continue the conversation in #28

**Pull request closed without merging**